pkg/registry/controller/strategy.go

```diff
@@ -92,9 +93,31 @@ func (rcStrategy) AllowCreateOnUpdate() bool {
 
 // ValidateUpdate is the default update validation for an end user.
 func (rcStrategy) ValidateUpdate(ctx api.Context, obj, old runtime.Object) field.ErrorList {
-	validationErrorList := validation.ValidateReplicationController(obj.(*api.ReplicationController))
-	updateErrorList := validation.ValidateReplicationControllerUpdate(obj.(*api.ReplicationController), old.(*api.ReplicationController))
-	return append(validationErrorList, updateErrorList...)
+	oldRc := old.(*api.ReplicationController)
```

Comment:
I don't understand this comment. What do we gain by having replica sets at this level? Won't we need to validate any non-convertible annotation? Should I use the custom conversion here and validate if I get back any error?